### PR TITLE
rocksdb: drop redundant `pin-to-gcc12-if-gcc13`

### PR DESCRIPTION
**Pull request by trofi**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by trofi** [comment]
Result of `nixpkgs-review pr 294800` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages failed to build:</summary>
  <ul>
    <li>blockbook</li>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph.dev</li>
    <li>ceph.doc</li>
    <li>libceph (ceph.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph.man</li>
    <li>electrs</li>
    <li>fulcrum</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>rippled</li>
    <li>rippled-validator-keys-tool</li>
    <li>rocksdb_6_23</li>
    <li>rocksdb_6_23.tools</li>
    <li>rocksdb_7_10</li>
    <li>rocksdb_7_10.tools</li>
    <li>samba4Full</li>
    <li>samba4Full.dev</li>
    <li>samba4Full.man</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>chickenPackages_5.chickenEggs.rocksdb</li>
    <li>matrix-conduit</li>
    <li>polkadot</li>
    <li>rocksdb</li>
    <li>rocksdb.tools</li>
    <li>solana-cli</li>
    <li>solana-validator</li>
    <li>sortmerna</li>
    <li>stalwart-mail</li>
    <li>surrealdb</li>
  </ul>
</details>

**Comment by trofi** [comment]
`rocksdb_6_23` and `rocksdb_7_10` need a build fix. Converting to a draft.

**Comment by trofi** [comment]
`rippled-validator-keys-tool` is the only (pre-existing) build failure. Ready for review.

**Review by marsam — APPROVED** [review]

File: pkgs/development/libraries/rocksdb/default.nix [hunk 1]
```diff
@@ -39,21 +39,6 @@ stdenv.mkDerivation (finalAttrs: {
...
 
-  env.NIX_CFLAGS_COMPILE = toString (lib.optionals stdenv.cc.isGNU [
-    "-Wno-error=deprecated-copy"
-    "-Wno-error=pessimizing-move"
-    # Needed with GCC 12
-    "-Wno-error=format-truncation"
-    "-Wno-error=maybe-uninitialized"
-  ] ++ lib.optionals stdenv.cc.isClang [
-    "-Wno-error=unused-private-field"
-    "-faligned-allocation"
```
Comment (marsam) [inline pkgs/development/libraries/rocksdb/default.nix:50]: I think `x86_64-darwin` still needs `"-faligned-allocation"`

Comment (trofi) [reply pkgs/development/libraries/rocksdb/default.nix:50]: Good point! Restored `"-faligned-allocation"`

**Review by marsam — APPROVED** [review]

**Pull request merged**